components/locale/macros/src/token_stream.rs

```diff
@@ -0,0 +1,99 @@
+use icu_locale::subtags;
+use tinystr::{TinyStr4, TinyStr8};
+
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+
+pub(crate) trait IntoTokenStream {
+    fn into_token_stream_string(self) -> String;
```

Comment:
Why `self` instead of `&self`?

Reply:
To stay aligned with `Into` convention which consumes the source when producing the output - https://doc.rust-lang.org/std/convert/trait.Into.html#tymethod.into

It's a helper function, used only during compilation.